cmake/Modules/FindCcache.cmake

```diff
@@ -1,16 +1,9 @@
 ## cf https://stackoverflow.com/questions/1815688/how-to-use-ccache-with-cmake
 ## leading to https://invent.kde.org/kde/konsole/merge_requests/26/diffs
-find_program(CCACHE_FOUND NAMES "sccache" "ccache")
-set(CCACHE_SUPPORT ON CACHE BOOL "Enable ccache support")
-if (CCACHE_FOUND AND CCACHE_SUPPORT)
-  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" # GNU is GNU GCC
-      OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    # without this compiler messages in `make` backend would be uncolored
-    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -fdiagnostics-color=auto")
-  endif()
-  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_FOUND})
```

Comment:
Are you sure we can remove this?
https://cliutils.gitlab.io/modern-cmake/chapters/features/utilities.html#ccache
https://github.com/llvm/llvm-project/blob/35684fa4bc8d2288d479cb8aa9d275b14bfefead/llvm/CMakeLists.txt#L286

Reply:
I don't know anything about Windows and how building works there, but https://cmake.org/cmake/help/latest/prop_gbl/RULE_LAUNCH_COMPILE.html makes it clear that RULE_LAUNCH_COMPILE is internal.

And the commit message of https://github.com/llvm/llvm-project/commit/0fad18c44a14b909a1aa47165021273d4c10a665 seems to indicate CMake 3.4 should be enough, the commit was reverted in https://github.com/llvm/llvm-project/commit/fff762d674c34c56d82f71250df2ac3652992049 because the build bots needed to be updated first.

Reply:
It's likely setting these properties [caused compile errors](https://github.com/TileDB-Inc/TileDB/actions/runs/11256903428/job/31299912175). They used to be applied only on the superbuild so they were not actually being used before.